File: framework/src/gmsec4/field/U64Field.cpp
/*
 * Copyright 2007-2021 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


#include <gmsec4/field/U64Field.h>
#include <gmsec4/internal/field/InternalU64Field.h>


using namespace gmsec::api::internal;


namespace gmsec
{
namespace api
{

U64Field::U64Field(const char* name, GMSEC_U64 value)
	: m_internal(new InternalU64Field(name, value))
{
	registerChild(m_internal);
}


U64Field::U64Field(const U64Field& other)
	: m_internal(new InternalU64Field(other.getName(), other.getValue()))
{
	registerChild(m_internal);
}


U64Field::~U64Field()
{
	delete m_internal;
}


GMSEC_U64 U64Field::getValue() const
{
	return m_internal->getValue();
}


const char* U64Field::toXML() const
{
	return m_internal->toXML();
}


const char* U64Field::toJSON() const
{
	return m_internal->toJSON();
}

} //namespace api
} //namespace gmsec

File: csharp/gmsec5/dox/field/U64Field.cs
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */

/* This file is used for generating Doxygen documentation; it only contains
   stubs of constructors and methods associated with the class.
*/


namespace GMSEC.API5 {

/// <summary>
/// Specialized Field class that can be used to store an unsigned 64-bit value.
/// </summary>

public class U64Field : Field {

  /// <summary>Creates a U64Field object with the given value</summary>
  ///
  /// <param name="name">The name of the field</param>
  /// <param name="value">The value to store in the field</param>
  /// <param name="isHeader">Use to indicate if the field is a header field (default is false)</param>
  ///
  /// <exception cref="GmsecException">Thrown if the name is null, is an empty string, or otherwise if the field name is non-compliant.</exception>
  public U64Field(string name, ulong value, bool isHeader = false) {
  }

  /// <summary>Copy constructor</summary>
  /// <param name="other">The other U64Field object to copy</param>
  public U64Field(U64Field other) {
  }

  /// <summary>Returns the value associated with the field</summary>
  ///
  /// <returns>The value associated with the field</returns>
  public ulong GetValue() {
  }

}

}
